query_execution/tests/DAGAnalyzer_unittest.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ **/
+
+#include <algorithm>
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "query_execution/DAGAnalyzer.hpp"
+#include "query_execution/tests/MockOperator.hpp"
+#include "utility/DAG.hpp"
+
+namespace quickstep {
+
+TEST(DAGAnalyzerTest, ZeroNodeDAGTest) {
+  std::unique_ptr<DAG<RelationalOperator, bool>> zero_node_dag;
+  zero_node_dag.reset(new DAG<RelationalOperator, bool>());
```

Comment:
Optionally, I suggest

`auto zero_node_dag  = std::make_unqiue<DAG<RelationalOperator, bool>>();`
or simply `DAG<RelationalOperator, bool> zero_node_dag;`, instead of two statements.